### ISSUE TITLE
Flame Error: The following assertion was thrown during performResize(): The GameRenderBox class does not implement "computeDryLayout".

### ISSUE THREAD

**Issue by YohanWadia** [flutter/flutter#73360]
This is the code. its the simplest example given by JapAlekhin... it just displays a Box in the middle of the screen that you can tap on...
It works fine... But the moment I change to beta channel and try to run it as a web game in chrome it throws this error.
This error is thrown for any game I try to run on the web ie - any game that runs in stable channel without any error, if I  change the channel to beta and run in chrome... I get this error.

could someone help me out to just be able to run a flame game in chrome? Because flutter needs me to be in beta for that, and somehow i keep getting this weird error on every game I run, when i try to run them in chrome via beta channel!?

### ERROR===========================
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
The following assertion was thrown during performResize():
The GameRenderBox class does not implement "computeDryLayout".
If you are not writing your own RenderBox subclass, then this is not
your fault. Contact support: https://github.com/flutter/flutter/issues/new?template=2_bug.md

When the exception was thrown, this was the stack:
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 216:49  throw_
packages/flutter/src/rendering/box.dart 1898:9                                                                             <fn>
packages/flutter/src/rendering/box.dart 1901:14                                                                            debugCannotComputeDryLayout
packages/flutter/src/rendering/box.dart 1858:12                                                                            computeDryLayout
packages/flutter/src/rendering/box.dart 2321:12                                                                            performResize
packages/flame/game/game_render_box.dart 26:11                                                                             performResize
packages/flutter/src/rendering/object.dart 1756:9                                                                          layout
packages/flutter/src/rendering/proxy_box.dart 281:7                                                                        performLayout
packages/flutter/src/rendering/object.dart 1777:7                                                                          layout
packages/flutter/src/rendering/proxy_box.dart 120:7                                                                        performLayout
packages/flutter/src/rendering/object.dart 1777:7                                                                          layout
packages/flutter/src/rendering/view.dart 167:7                                                                             performLayout
packages/flutter/src/rendering/object.dart 1634:7                                                                          [_layoutWithoutResize]
packages/flutter/src/rendering/object.dart 884:17                                                                          flushLayout
packages/flutter/src/rendering/binding.dart 454:19                                                                         drawFrame
packages/flutter/src/widgets/binding.dart 900:13                                                                           drawFrame
packages/flutter/src/rendering/binding.dart 320:5                                                                          [_handlePersistentFrameCallback]
packages/flutter/src/scheduler/binding.dart 1119:15                                                                        [_invokeFrameCallback]
packages/flutter/src/scheduler/binding.dart 1057:9                                                                         handleDrawFrame
packages/flutter/src/scheduler/binding.dart 865:7                                                                          <fn>
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/isolate_helper.dart 48:19       internalCallback

The following RenderObject was being processed when the exception was fired: GameRenderBox#af7e0 NEEDS-LAYOUT NEEDS-PAINT:
  parentData: <none> (can use size)
  constraints: BoxConstraints(w=500.0, h=932.0)
  size: MISSING
This RenderObject has no descendants.
════════════════════════════════════════════════════════════════════════════════════════════════════
Another exception was thrown: RenderBox did not set its size during layout.
Another exception was thrown: Assertion failed: file:///C:/flutterio/flutter/packages/flutter/lib/src/rendering/box.dart:1920:12
Another exception was thrown: Assertion failed: file:///C:/flutterio/flutter/packages/flutter/lib/src/rendering/box.dart:1920:12
Another exception was thrown: Assertion failed: file:///C:/flutterio/flutter/packages/flutter/lib/src/rendering/box.dart:1920:12

===============================

```
void main() async{
  WidgetsFlutterBinding.ensureInitialized();

  Util flameUtil = Util();
  await flameUtil.fullScreen();
  await flameUtil.setOrientation(DeviceOrientation.portraitUp);

  Flame.images.loadAll(<String>[
    'bg/backyard.png',
    
  ]);
  
  Flame.audio.disableLog();
  Flame.audio.loadAll(<String>[    
    'bgm/playing.mp3',
  ]);


  BoxGame game = BoxGame(); 
  TapGestureRecognizer tapper = TapGestureRecognizer();
  tapper.onTapDown = game.onTapDown;
  runApp(game.widget);
  flameUtil.addGestureRecognizer(tapper);
}
```

```
class BoxGame extends Game {
  Size screenSize;
  bool hasWon=false;

  BoxGame(){
    initialize();
  }
  void initialize() async {
    Size initSize = await Flame.util.initialDimensions();
    resize(initSize);
  }

  void render(Canvas canvas) {
    Rect bgRect = Rect.fromLTWH(0, 0, screenSize.width, screenSize.height);
    Paint bgPaint = Paint();
    bgPaint.color = Color(0xffffffff);
    canvas.drawRect(bgRect, bgPaint);

    double screenCenterX = screenSize.width / 2;
    double screenCenterY = screenSize.height / 2;
    Rect boxRect = Rect.fromLTWH(
        screenCenterX - 75,
        screenCenterY - 75,
        150,
        150
    );
    Paint boxPaint = Paint();



    if (hasWon) {
      //print("1111111");
      boxPaint.color = Color(0xff00ff00);
    } else {
      //print("00000000000");
      boxPaint.color = Color(0xff341090);
    }
    canvas.drawRect(boxRect, boxPaint);
  }

  void update(double t) {

  }

  void resize(Size size) {
    screenSize = size;
    super.resize(size);
  }

  void onTapDown(TapDownDetails d) {
    double x= d.globalPosition.dx;
    double y= d.globalPosition.dy;
    print("CLICKED.....$x|$y");

    double screenCenterX = screenSize.width / 2;
    double screenCenterY = screenSize.height / 2;
    if (x >= screenCenterX - 75
        && x <= screenCenterX + 75
        && y >= screenCenterY - 75
        && y <= screenCenterY + 75
    ) {
      hasWon = !(hasWon);
    }


  }

}
```

**Comment by christopherfujino** [comment]
This is an issue with the Flame package, not Flutter, and it was fixed upstream https://github.com/flame-engine/flame/issues/586 and https://github.com/flame-engine/flame/pull/587

**Comment by YohanWadia** [comment]
I dint understand "it was fixed upstream flame-engine/flame#586 and flame-engine/flame#587" ???

I cant find a solution out there.... and if it were fixed im still getting it!?! any advise?

**Comment by christopherfujino** [comment]
please follow up on the flame issue tracker.
